lib/internal/streams/duplexify.js

```diff
@@ -85,6 +85,11 @@ module.exports = function duplexify(body, name) {
   if (typeof body === 'function') {
     const { value, write, final, destroy } = fromAsyncGen(body);
 
+    // Body might be a constructor function instead of an async generator function.
+    if (isDuplexNodeStream(value)) {
```

Comment:
should we reflect this in the docs?

Reply:
Hard to say, since this is for internal use and the way of init is not decently

Reply:
It's used for public API `Duplex.from`, whose behavior will also be slightly changed

https://github.com/nodejs/node/blob/356b4a268a6f20c11dc79ed841c53020c00b8872/lib/internal/streams/duplex.js#L142-L147

Reply:
Duplex.from is still an experimental API. I will put it into doc later

Reply:
Stable `pipeline` also affected

https://github.com/nodejs/node/blob/4032ad593c0dfda2b57266339b9ab6b1b3c9648e/lib/internal/streams/pipeline.js#L288-L298

Reply:
I will update doc later